### ISSUE TITLE
v4.8.3 - no Auto-deletion or Retention Settings for Audio Recordings

### ISSUE THREAD

**Issue by rkhrkh** [braden-w/whispering#285]
Currently, the app (Whispering 4.8.3 for aarch64) lacks an option in the Settings to configure the retention period for previous recordings or to disable history recordings altogether. 

This limitation results in the accumulation of numerous audio files, which users must delete manually on a regular basis.

**Feature request:** Please implement a setting that allows users to:

1. Specify how many days of previous recordings to keep (e.g., 2 days, 30 days, etc.)
2. Automatically delete audio files older than the specified retention period
3. Optionally disable history recordings entirely

This feature would improve user experience by reducing manual maintenance and storage management tasks.

**Comment by braden-w** [comment]
Great feature request, will take a look into this when I can!

**Comment by braden-w** [comment]
Sorry @rkhrkh for the delay—I've been in training for school for the past few days and it got really intense. Thank you for the resubscribe! I plan on getting to this tomorrow. Hope everything has been going well!

**Comment by rkhrkh** [comment]
**This issue has now escalated** into a significant problem. In the past, I regularly deleted older audio files whenever I remembered to do so. However, I haven't done this for quite some time. Now, when I try to view the recordings, it takes several minutes for the application to respond. During this time, the application freezes completely. 

After these minutes pass, the recordings do not appear as expected; instead, they are **displayed as empty entries.** I have included an image to illustrate this issue.

If this problem persists, I will need to take action to prevent the accumulation of too many recordings. My solution would be to revert to an older version of the application where this issue does not occur.
<img width="422" alt="image" src="https://github.com/user-attachments/assets/ea4e6347-4764-4e80-8f99-57de39fcf7a9" />
